### PR TITLE
chore: add Technology-HuaweiCloud repository

### LINKED ISSUE #4154: Adoption Request
# Adoption Request

This adoption request aims at contributing a new technology feature pack in the form of a new Technology Repository. The contributing company is Huawei Technologies Co., Ltd.

__This Adoption Request is a follow-up on the first one submitted 22nd of November 2023. Back then, the Committer of EDC requested Huawei to show those components are maintained accordingly to EDC releases and the components are adopted in industry.  Both aspects were presented in a Committer meeting on 17th of April, which leads to a positive vote for creating the new repository for Huawei's extensions in EDC.__

The contribution initially will contain two major feature areas:
- OBS: S3-compatible object storage hosted in the cloud
- GaussDB: a relational database based on Postgres 9.2.4 as data retention backend

These components are currently implemented using EDC SPIs and core features. 

## General information

Please provide some information about your project or code contribution. 

| Title | Description | Contact | Links
| :--- | :--- | :--- | :---
| Huawei Dataspace Components | Technology repository named "Technology-Huawei" containing OBS extensions and GaussDB extensions | atul.athavale@huawei.com, chenliang613@huawei.com, swaroop.akulu.raghupathy@huawei.com   | https://github.com/Huawei-Dataspace-Components/components

## Adoption level

Next, please tell us what level of adoption you intend to provide.

- [ ] Reference a feature as "friend"
- [x] Incorporate a feature as core EDC component

## Adoption in EDC core

During an initial ramp-up phase, Huawei Technologies Co., Ltd. contracts existing committers of the EDC Technical Committers group to maintain and develop the code base. This phase will be supported and accompanied by two Huawei employees. Over time, Huawei Technologies Co., Ltd. will build up an internal team to take over maintenance of the `Technology-Huawei` repository to ensure the rationale mentioned in [Decision: Separate build pipelines for Technology Repositories](https://github.com/eclipse-edc/docs/tree/main/developer/decision-records/2024-04-19-tech-repo-split).

Huawei Technologies Co., Ltd. commits to provide real cloud test infrastructure, if no self-contained (containerized) variant is available.

### PR DESCRIPTION
**Pull request by mspiekermann**
## What this PR changes/adds

This PR adds the new Technology-HuaweiCloud repository to the EDC-Org.

## Why it does that

According to the Adoption Request eclipse-edc/Connector#4154 we will add a new technology feature pack in the form of a new Technology Repository. The contributing company is Huawei Technologies Co., Ltd.

## Linked Issue(s)

Closes eclipse-edc/Connector#4154

### CODE REVIEW

File: otterdog/eclipse-edc.jsonnet [hunk 1]
```diff
@@ -458,6 +458,29 @@ orgs.newOrg('eclipse-edc') {
...
+    orgs.newRepo('Technology-HuaweiCloud') {
+      allow_rebase_merge: false,
+      allow_update_branch: false,
+      delete_branch_on_merge: false,
+      has_discussions: true,
+      has_wiki: false,
+      squash_merge_commit_title: "PR_TITLE",
+      web_commit_signoff_required: false,
+      workflows+: {
+        default_workflow_permissions: "write",
```
Comment (netomi) [inline otterdog/eclipse-edc.jsonnet]: it looks like you copied an existing repo, which is perfectly fine, though for new repos we would like to already use default workflow permissions set to read as this should become the default for all repos ideally at some point in time. While workflows in existing repos might be some effort, creating new workflows in new repos should already default to the new fine grained permission model that GitHub supports imho.

File: otterdog/eclipse-edc.jsonnet [hunk 2]
```diff
@@ -458,6 +458,29 @@ orgs.newOrg('eclipse-edc') {
...
+      },
+      secrets: [
+        orgs.newRepoSecret('DISCORD_WEBHOOK_GITHUB') {
```
Comment (netomi) [inline otterdog/eclipse-edc.jsonnet]: are these secrets required? if so, we need to resolve them with references to our credential store. The JENKINS token is easy to setup but the discord webhook will need to be setup by the project team and then shared with us via a secure channel.

We could also start without secrets and add them as needed.

Comment (mspiekermann) [reply otterdog/eclipse-edc.jsonnet]: We don't need the discord webhook anymore...

**Comment by netomi** [comment]
LGTM, I will amend the change with the secrets.

File: otterdog/eclipse-edc.jsonnet [hunk 3]
```diff
@@ -457,6 +457,29 @@ orgs.newOrg('eclipse-edc') {
...
+          value: "********",
+        },
+        orgs.newRepoSecret('EF_JENKINS_TOKEN') {
+          value: "********",
+        },
+        orgs.newRepoSecret('EF_JENKINS_USER') {
+          value: "********",
+        },
```
Comment (fredg02) [inline otterdog/eclipse-edc.jsonnet]: @mspiekermann this requires more context. What are you trying to do with the EF_JENKINS secrets?

Comment (mspiekermann) [reply otterdog/eclipse-edc.jsonnet]: As with the other Technology-XYZ repositories, this new repository is part of the overall release process and thus during the workflow need those secrets.

Comment (netomi) [reply otterdog/eclipse-edc.jsonnet]: could this be outdated? I checked where the EF_JENKINS_TOKEN is being used, and the only reference is here:

https://github.com/eclipse-edc/.github/blob/86b7c9cc416398c6f497141bf69ad4265361a247/.github/workflows/_trigger-snapshot.yml#L14

this does not seem to use the passed jenkins_token at all.

There was a commit some time ago that added publishing directly from GitHub Actions:

https://github.com/eclipse-edc/.github/commit/ec430c3470c5bf7efd2976f685b6d88cad56ef90

maybe in this refactoring it was forgotten to cleanup unused secrets?

Comment (mspiekermann) [reply otterdog/eclipse-edc.jsonnet]: @netomi You are right, we can indeed remove those secrets. Nevertheless, as I would then remove them also for all the other repositories, will create a separate issue and PR for this.

Comment (netomi) [reply otterdog/eclipse-edc.jsonnet]: ok I removed them from the new repo already as it does not make sense to add it first and then remove again.

**Comment by netomi** [comment]
as this PR does not remove anything, you can auto-merge at your convenience.

**Comment by mspiekermann** [comment]
@netomi I can't because it is blocked until one of you two approves.

**Comment by netomi** [comment]
You can comment with /otterdog merge to merge if its eligible which is the case.

**Pull request merged**